proctor-common/src/main/java/com/indeed/proctor/common/RuleEvaluator.java

```diff
@@ -159,7 +162,59 @@ public boolean evaluateBooleanRuleWithValueExpr(
                         + " from rule "
                         + rule);
     }
+    /**
+     * @deprecated Use evaluateBooleanRulePartialWithValueExpr(String, Map) instead, it's more
+     *     efficient
+     */
+    @Deprecated
+    public boolean evaluateBooleanRulePartial(
+            final String rule, @Nonnull final Map<String, Object> values)
+            throws IllegalArgumentException {
+        final Map<String, ValueExpression> localContext =
+                ProctorUtils.convertToValueExpressionMap(expressionFactory, values);
+        return evaluateBooleanRulePartialWithValueExpr(rule, localContext);
+    }
 
+    public boolean evaluateBooleanRulePartialWithValueExpr(
```

Comment:
Can you add some javadoc for this method. I think we should make it clear in the docs that this should not be used except for your use case.